src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,40 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
```

Comment:
The method 'convertToTransaction' seems to be doing too much. It's parsing the data, validating it, and creating 'FruitTransaction' objects. Consider refactoring to follow the Single Responsibility Principle (SRP) and make the code more maintainable.